src/infra/dataset-update-flow-inmem/src/services/flow/flow_service_inmem.rs

```diff
@@ -0,0 +1,733 @@
+                            // TODO: throttle not from NOW,
+                            //  but from last flow of the dependent daataset
+                            let now = self.time_source.now();
+                            self.enqueue_flow(
+                                dependent_dataset_flow.flow_id,
+                                now + throttling_period,
+                            )?;
+
+                            dependent_dataset_flow
+                                .define_start_condition(
+                                    now,
+                                    FlowStartCondition::Throttling(FlowStartConditionThrottling {
+                                        interval: throttling_period,
+                                    }),
+                                )
+                                .int_err()?;
+                        }
+
+                        dependent_dataset_flow
+                            .save(self.flow_event_store.as_ref())
+                            .await
+                            .int_err()?;
+                    }
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    fn find_pending_flow(&self, flow_key: &FlowKey) -> Option<FlowID> {
+        let state = self.state.lock().unwrap();
+        state.pending_flows.try_get_pending_flow(flow_key)
+    }
+
+    async fn make_new_flow(
+        &self,
+        flow_key: FlowKey,
+        trigger: FlowTrigger,
+    ) -> Result<Flow, InternalError> {
+        let flow = Flow::new(
+            self.time_source.now(),
+            self.flow_event_store.new_flow_id(),
+            flow_key,
+            trigger,
+        );
+
+        let mut state = self.state.lock().unwrap();
+        state
+            .pending_flows
+            .add_pending_flow(flow.flow_key.clone(), flow.flow_id);
+
+        Ok(flow)
+    }
+
+    async fn merge_secondary_flow_trigger(
+        &self,
+        flow_id: FlowID,
+        trigger: FlowTrigger,
+    ) -> Result<FlowState, InternalError> {
+        let mut flow = Flow::load(flow_id, self.flow_event_store.as_ref())
+            .await
+            .int_err()?;
+        flow.add_trigger(self.time_source.now(), trigger)
+            .int_err()?;
+        flow.save(self.flow_event_store.as_ref()).await.int_err()?;
+        Ok(flow.into())
+    }
+
+    fn enqueue_flow(
+        &self,
+        flow_id: FlowID,
+        activation_time: DateTime<Utc>,
+    ) -> Result<(), InternalError> {
+        self.state
+            .lock()
+            .unwrap()
+            .time_wheel
+            .activate_at(activation_time, flow_id)?;
+        Ok(())
+    }
+
+    async fn schedule_flow_task(&self, flow: &mut Flow) -> Result<(), InternalError> {
+        let logical_plan = match &flow.flow_key {
+            FlowKey::Dataset(flow_key) => match flow_key.flow_type {
+                DatasetFlowType::Update => LogicalPlan::UpdateDataset(UpdateDataset {
```

Comment:
Should we have something like `Probe` for the flow system too, so we could test it manually without spawning expensive tasks?

Perhaps a `Probe` flow could run `Probe` tasks on a certain dataset and spawn `Probe` flows for all downstream, like a cascading update but the actual work will be a no-op?

Reply:
I would likely introducing something like this when testing.

Reply:
There is no mechanism to pass any parameters to the flow yet. This would be necessary for `Probe` idea to make sense.